src/plugins/link/main/ts/core/Utils.ts

```diff
@@ -133,6 +147,8 @@ const link = (editor: Editor, attachState: AttachState, data: LinkDialogOutput)
       linkAttrs.rel = newRel ? newRel : null;
     }
 
+    linkAttrs.href = handleExternalTargets(linkAttrs.href, Settings.forceExternalTargets(editor.settings));
```

Comment:
This runs after `DialogConfirms` (see https://github.com/tinymce/tinymce/blob/master/src/plugins/link/main/ts/ui/Dialog.ts#L51) so the dialog still shows up in your test

Reply:
Now that `AssumeExternalTargets` is an `Enum`, the dialog is opened when `AssumeExternalTargets` is either `true` or `false`, not for `'http'` and `'https'`.